### DISCUSSION TITLE
Feature Request: Automatically load a mapping preset based on connected controller

### DISCUSSION THREAD

**Discussion by NeitherMeNorYou** [darthcloud/BlueRetro#844] [Ideas]
Greetings!

I got a GC Retrotime Blueretro adapter from 8bitmods recently, and have been liking it so far. However what I found to be a bit annoying is that there is no way to comfortably use multiple differently styled controllers with it without manually having to change the key mapping every time.
For me specifically I want to use a Switch SNES controller for my GB Player (and potentially some other simple platformers), and an 8bitdo Ultimate for pretty much anything else. While I could create a GameID profile for all the GB Player apps in SWISS (the actual start up disc + GB Interface) that would still leave the SWISS environment with a key map that doesn't match the print on the SNES controller that I would prefer. However if the key map could be loaded on connection of the controller the entire process of going through SWISS and then in the GB Player would be a lot more seamless.

Ideally this would work nearly identical to GameID in that it gets stored as a preset in the file system, and maybe with a toggle in the web interface to prevent GameID from loading if a ControllerID (trademark pending) preset is already loaded. With roughly 12 slots available for GameID a few of those could be used for ControllerIDs to suit people's needs so people could still mix and match them as desired.

**Comment by darthcloud** [comment]
Yeah that something I want to add in the future. But probably not soon.

**Comment by NeitherMeNorYou** [comment]
That'd be awesome, thank you!

In the meantime I'll have a look if I can't find a solution with tampermonkey or something to make it a little faster to swap on the web app.
